oauth-common/src/main/java/io/strimzi/kafka/oauth/services/OAuthMetrics.java

```diff
@@ -90,26 +111,48 @@ private Metrics initKafkaMetrics() {
 
     private List<MetricsReporter> initReporters() {
         AbstractConfig kafkaConfig = initKafkaConfig();
-        List<MetricsReporter> reporters = kafkaConfig.getConfiguredInstances(CommonClientConfigs.METRIC_REPORTER_CLASSES_CONFIG,
-                MetricsReporter.class);
-
-        JmxReporter reporter = new JmxReporter();
-        reporter.configure(configMap);
-
-        reporters.add(reporter);
+        List<MetricsReporter> reporters = new ArrayList<>(3);
+        if (configMap.get(STRIMZI_METRIC_REPORTERS) != null) {
+            reporters = kafkaConfig.getConfiguredInstances(STRIMZI_METRIC_REPORTERS, MetricsReporter.class);
+        } else {
+            log.warn("Configuration option '{}' is not set, OAuth metrics will not be exported to JMX", STRIMZI_METRIC_REPORTERS);
```

Comment:
We define the config as low importance below and without a default value but if it's not set we emit a warning. Should we increase the importance or downgrade that log level?

Reply:
I don't know how you decide on the importance of the particular configuration option. In this case the warning is most welcome because it is almost certainly an omission that STRIMZI_METRIC_REPORTERS was not configured. With metric enabled and not exported to JMX there is no way to collect by any mechanism (as fas as I know).

Reply:
If not setting it generates warnings, it looks pretty important to me. That said I'm not sure what conventions are used in other parts of the Strimzi code so if none of the maintainers pick on this, maybe we can ignore it.

Reply:
I would agree with Mickael that this seems a bit weird. I guess when you put it that enabling metrics but not configuring the metrics reporter would be wrong, that makes sense. But why does the user need to:
* Enable the metrics
* And configure the metrics reporter

Why not do just one and have the other done automatically by the Oauth library? I guess I'm missing some obvious reason why it is not possible, but it would be good to understand it.

Reply:
For OAuth metrics I didn't want them to be enabled by default in order to avoid potential issues (performance or memory consumption for example). And that did allow us to very easily work around CruiseControl issue - everything was fine until someone enabled OAuth metrics. The solution was - don't enable OAuth metrics.

OAuth metrics also has finer control on metrics collection. You can enable metrics globally, or for just particular listeners, or just for KeycloakAuthorizer.

Kafka metrics are enabled by default and which of them are enabled is controlled by [metrics.recording.level](https://kafka.apache.org/documentation/#producerconfigs_metrics.recording.level). But as far as I understand there is no way to completely turn them off.

For OAuth we could change the default for the configuration (enabled by default and you can disable it), but if we otherwise keep  backwards compatibility around `metric.reporters` this will result in things crashing / acting up by default when non-compatible metric reporters (such as CruiseControl metric reporter) are configured, as we would double-instantiate them.
We could then enable metrics gathering only when some reporters are configured via 'strimzi.oauth.metric.reporters'. However, if we made this change then absence of `strimzi.oauth.metric.reporters` if we want to maintain backwards compatibility with `metrics.reporters` would mean that we have to instantiate `JmxReporter` each and every time even if `metric.reporters` is not set.

I would prefer for now to NOT have OAuth metrics enabled by default. Let's have it opt-in until we're confident that there are no issues when using them (like the issue this PR is addressing of things crashing when a reporter not compatible with OAuth metrics is configured).

Reply:
I can understand why OAuth metrics should not be enabled by default. But why does the user need to configure the `strimzi.oauth.metric.reporters`? Isn't it Strimzi OAuth who consumes this option anyway? So why not set it automatically.

Reply:
Ah, yes that's a great point. I was motivated by aligning with where [Kafka's handling of `metric.reporters`](https://cwiki.apache.org/confluence/display/KAFKA/KIP-830%3A+Allow+disabling+JMX+Reporter) is headed.
With Kafka 4.0 there will be no exporting of (automatically enabled) metrics to JMX unless a specific reporter is configured that provides such an export.

We can opt to not go this way to avoid needing to set two configurations (Kafka does not have that problem since its metrics are always enabled). We could for example handle omission of `strimzi.oauth.metric.reporters` by automatically creating JmxReporter. If user specifies `strimzi.oauth.metric.reporters` we could only instantiate the configured reporters, but we would not check if JmxReporter is part of the config, and we would not automatically instantiate it if not present in the config (that's different from current `metric.reporters` handling when we automatically instantiate JmxReporter if it is not part of the `metric.reporters` list).

That probably makes more sense in our case as then it is not necessary to always set it, and user should consult the documentation anyway to determine the behaviour and that it is not the same as for Kafka metrics configured via `metric.reporters`.
Also the presence of `strimzi.oauth.metric.reporters=` (configured and set to empty value) could then mean to NOT instantiate a JmxReporter if for some reason one really wants to have no reporters attached to the OAuth Metrics object (and thus no export to JMX).

Reply:
Also keep in mind that hopefully that should all disappear in the near future once KIP-877 gets in Kafka. So maybe we want to keep things simple.

Reply:
Ok, I'll modify the logic as described above. It means no backwards compatibility in the sense that `metric.reporters` will be ignored. But if `strimzi.oauth.metric.reporters` is set it will be used, if not set, `JmxReporter` will automatically be instantiated.